code/game/objects/items/hot_potato.dm

```diff
@@ -52,6 +52,9 @@
 	var/atom/location = loc
 	location.visible_message(span_userdanger("[src] [detonate_explosion? "explodes" : "activates"]!"), span_userdanger("[src] activates! You've ran out of time!"))
 	if(detonate_explosion)
+		var/mob/living/m = loc
+		if(m.get_active_held_item()==src)
```

Comment:
It's actually possible that the hot potato isn't located inside a living mob, so please include some istype() check at the start of this conditional statement..